diff --git a/src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java b/src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java
index 014ff920..dd13dd35 100644
--- a/src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java
+++ b/src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java
@@ -156,6 +156,7 @@ public class TimestampIncrementingTableQuerier extends TableQuerier implements C
     addSuffixIfPresent(builder);
     
     String queryString = builder.toString();
+    queryString = "SELECT * FROM ( " + queryString + " )"
     recordQuery(queryString);
     log.trace("{} prepared SQL query: {}", this, queryString);
     stmt = dialect.createPreparedStatement(db, queryString);
