diff --git a/src/main/java/io/confluent/connect/jdbc/source/BulkTableQuerier.java b/src/main/java/io/confluent/connect/jdbc/source/BulkTableQuerier.java
index 2f2d0613..4d7cd082 100644
--- a/src/main/java/io/confluent/connect/jdbc/source/BulkTableQuerier.java
+++ b/src/main/java/io/confluent/connect/jdbc/source/BulkTableQuerier.java
@@ -58,7 +58,7 @@ public class BulkTableQuerier extends TableQuerier {
 
         break;
       case QUERY:
-        builder.append(query);  
+        builder.append("SELECT * FROM (").append(query).append(")");  
         
         break;
       default:
